spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/jackson/FieldDocumentationVisitorContext.java

```diff
@@ -106,15 +111,28 @@ private Attribute optionalAttribute(Class<?> javaBaseClass, String javaFieldName
     }
 
     private List<String> resolveConstraintDescriptions(Class<?> javaBaseClass,
-            String javaFieldName) {
-        List<String> descriptions = constraintReader
-                .getConstraintMessages(javaBaseClass, javaFieldName);
+            String javaFieldName, JavaType javaType) {
+        List<String> descriptions = new ArrayList<>();
+        descriptions.addAll(constraintReader.getConstraintMessages(javaBaseClass, javaFieldName));
 
         // fallback to field itself if we got a getter and no annotation on it
         if (descriptions.isEmpty() && isGetter(javaFieldName)) {
-            descriptions = resolveConstraintDescriptions(javaBaseClass, fromGetter(javaFieldName));
+            descriptions.addAll(resolveConstraintDescriptions(javaBaseClass,
+                    fromGetter(javaFieldName), javaType));
+        }
+
+        if (javaType.isEnumType()) {
+            String enumDesc = enumConstraintDescription((Class<Enum>) javaType.getRawClass());
+            // we might be recursively here, prevent duplicate add
+            if (!descriptions.contains(enumDesc)) {
+                descriptions.add(enumDesc);
+            }
         }
 
         return descriptions;
     }
+
+    private String enumConstraintDescription(Class<Enum> rawClass) {
+        return "Must be one of " + arrayToString(rawClass.getEnumConstants());
```

Comment:
It would be better if a client of the library can configure the text. All other texts can be changed if required. This would allow for documentation in other languages than English. But you do not have to do the change in this PR.